src/macos/cpid_macos.c

```diff
@@ -0,0 +1,288 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+
+#include <IOKit/IOKitLib.h>
+#include <sys/sysctl.h>
+#include <openssl/evp.h>
+
+#include "cpid/cpid_macos.h"
+#include "../constants.h"
+
+#define KERNEL_TASK_PID 0
+#define LAUNCHD_PID 1
+#define MACOS_SERIAL_NUMBER_BUFFER_SIZE 16
+#define MACOS_EXPECTED_DIGEST_INPUT_CONTENT_SIZE 88
+#define MAX_MICROS_OFFSET 999999
+#define OPEN_SSL_SUCCESS 1
+
+#pragma pack(push, 1)
+typedef struct {
+    uint64_t unix_epoch_seconds;
+    uint64_t micros_offset;
+} process_creation_time_t;
+
+typedef struct {
+    char serial_number[MACOS_SERIAL_NUMBER_BUFFER_SIZE]; // 16 bytes
+    uuid_t hardware_uuid; // 16 bytes
+    process_creation_time_t kernel_task_creation_time;
+    process_creation_time_t launchd_creation_time;
+    process_creation_time_t process_creation_time;
+    uint64_t pid;
+} digest_input_content_t;
+
+#pragma pack(pop)
+
+_Static_assert(MACOS_EXPECTED_DIGEST_INPUT_CONTENT_SIZE == sizeof(digest_input_content_t), "digest_input_content_t is not the expected size.");
+
+typedef struct cpid_struct {
+    EVP_MD_CTX *digest_context;
+    EVP_MD *sha256;
+    uint8_t digest_destination_buffer[EVP_MAX_MD_SIZE];
+    digest_input_content_t digest_input_content;
+} *cpid_handle_t;
```

Comment:
As noted above, I don't think you should declaratively associate `cpid_handle_t` and `cpid_struct`.